komorebi-bar/src/group.rs

```diff
@@ -0,0 +1,161 @@
+use crate::config::AlphaColour;
+use eframe::egui::Color32;
+use eframe::egui::Frame;
+use eframe::egui::InnerResponse;
+use eframe::egui::Margin;
+use eframe::egui::Rounding;
+use eframe::egui::Stroke;
+use eframe::egui::Ui;
+use komorebi_client::Colour;
+use komorebi_client::Rect;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use serde::Serialize;
+
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, JsonSchema)]
+pub enum Grouping {
```

Comment:
Let's make this `#[serde(tag = "kind")]` so that it's possible to specify

```json
{
    "grouping": {
        "kind": "Widget",
        "fill": ...,
        "rounding: ...,
        "outer_margin" ...,
        // etc
    }
}
```

in the same way that we can distinguish between the Catppuccin and Base16 `theme` options by `palette`

Reply:
That looks better 👍